lib/monitoring/index.js

```diff
@@ -0,0 +1,27 @@
+const {
+  LogLevel,
+  LogEvent,
+  logNothing,
+  AbstractLoggingEngine,
+  logToConsole
+} = require('./logging-engine');
+
+const configureLogging = (engine) => (system) => {
+  if (system.loggingEngine) {
+    throw new Error('Do not configure logging more than once. The behavior could be undefined otherwise.');
+  }
+
+  if (!(engine instanceof AbstractLoggingEngine)) {
```

Comment:
@iskandersierra I initially was doing type checks, but it turns out it causes problems with plugins such as the postgres persistence engine: It forces all dependent libraries to upgrade their nact versions in lock step.

Reply:
Fair. I changed it by:

``` javascript
  if (!engine) {
    throw new Error('Expected a LoginEngine.');
  }
```

Any comments on the previous issue on configureLogging?

Reply:
@iskandersierra I've merged an update which should fix the error below:
``********************** system reference type is: ActorSystemReference
********************** Error getting logging engine: Error: Actor has stopped or never even existed```

The issue was that the system was being added to the system map AFTER the extensions were being configured.

Reply:
Great! Now it is failing further away, in this code:
> TypeError: Cannot read property 'system' of undefined
    at Object.module.exports.applyOrThrowIfStopped (C:\Dev\git\nact\lib\system-map.js:1:3402)
    at spawn (C:\Dev\git\nact\lib\actor.js:2:3509)
    at spawnStateless (C:\Dev\git\nact\lib\actor.js:2:3856)
    at spawnLoggingActor (C:\Dev\git\nact\lib\monitoring\logging-actor.js:1:3940)
    at system (C:\Dev\git\nact\lib\monitoring\index.js:1:3098)
    at C:\Dev\git\nact\test\logging-actor.js:39:40
``` javascript
const spawn = (parent, f, name, properties) =>
  systemMap.applyOrThrowIfStopped(parent, p => p.assertNotStopped() && new Actor(p, name, p.system, f, properties).reference);
```

It seems that in ```p.system```, p is giving me undefined.

Reply:
Surely p.assertNotStopped() would also fail in that case?